### PR TITLE
feat: Add test reports folder to gitignore

### PR DESCRIPTION
**Pull request by ayush3160**
## Related Issue
Closes: #1775 

#### Describe the changes you've made

I have implemented a functionality in this pr to automatically add test reports folder to .gitignore file. When we will run test command then it creates a .gitignore file in keploy folder if not already present and add test report folder in it. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added
NIL
#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

## Screenshots (if any)
NA

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -760,3 +760,37 @@ func getHomeDir() (string, error) {
 func IsDockerCmd(kind CmdType) bool {
 	return (kind == DockerRun || kind == DockerStart || kind == DockerCompose)
 }
+
+func CreateGitIgnore(logger *zap.Logger, path string) error {
+	gitignorePath := path + "/.gitignore"
+	reportEntry := "/reports/"
```
Comment (charankamarapu) [inline utils/utils.go:766]: Be specific with the entry I think keploy/reports/ works better I guess. Please have a look.

**Review by charankamarapu — CHANGES_REQUESTED** [review]
Please address the comments, rest looks good. Let me get back to you about why the pipeline is failing

File: utils/utils.go (hunk 1 above)
Comment (ayush3160) [reply utils/utils.go:766]: I am storing .gitignore file in keploy folder itself as it was mentioned in the issue , So keploy/reports/ entry will not work.

Comment (charankamarapu) [reply utils/utils.go:766]: cool got it

**Comment by charankamarapu** [comment]
I think the action is failing because the ,gitignore file is also considered to be test set in the action . You can add the gitignore file like how report folder is added and I guess then it should work - https://github.com/keploy/keploy/blob/c306030272404fc19b2a1a8783ea1ceefd123154/.github/actions/tester/install.sh#L55C77-L55C86.

**Comment by ayush3160** [comment]
@charankamarapu , I think this is not the problem because when I ran the same command in terminal it is giving directories of test folder only as it should return. Please refer to the below Image. Let me know If I am doing something wrong here.

Screenshot from 2024-07-27 15-35-08

**Comment by charankamarapu** [comment]
hmm looks like thats not the problem. Let me deep dive more.

**Comment by charankamarapu** [comment]
Got the issue. We should consider only directories here in the readSession function. 
https://github.com/keploy/keploy/blob/c306030272404fc19b2a1a8783ea1ceefd123154/pkg/util.go#L199
the pilot which we are using for testing is using this function, due to this there is a mismatch in no of test sets. 
https://github.com/keploy/pilot/blob/c8c1d1a671385f4401c3f30f71b0888832787cac/main.go#L70

please raise separate PR for readSession fix and I review and merge it then and release, we can upgrade the keploy version in the pilot repo and that should fix it.

**Comment by ayush3160** [comment]
Thanks @charankamarapu , I will create separate pr for this.

**Comment by charankamarapu** [comment]
create PR on this repo by changing keploy version in the mod file - https://github.com/keploy/pilot

**Review by charankamarapu — APPROVED** [review]
LGTM

**Pull request merged**